src/internal/packager/images/pull_test.go

```diff
@@ -84,6 +85,54 @@ func TestCheckForIndex(t *testing.T) {
 }
 
 func TestPull(t *testing.T) {
+	t.Parallel()
+	testCases := []struct {
+		name        string
+		ref         string
+		expectedErr string
+	}{
+		{
+			name:        "pull an image",
+			ref:         "ghcr.io/zarf-dev/zarf/agent:v0.32.6@sha256:b3fabdc7d4ecd0f396016ef78da19002c39e3ace352ea0ae4baa2ce9d5958376",
+			expectedErr: "",
+		},
+		{
+			name:        "error when pulling an image that doesn't exist",
+			ref:         "ghcr.io/zarf-dev/zarf/imagethatdoesntexist:v1.1.1",
+			expectedErr: "No such image",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			ref, err := transform.ParseImageRef(tc.ref)
+			require.NoError(t, err)
+			destDir := t.TempDir()
+			pullConfig := PullConfig{
+				DestinationDirectory: destDir,
+				ImageList: []transform.Image{
+					ref,
+				},
+			}
+
+			pulled, err := Pull(context.Background(), pullConfig)
+			if tc.expectedErr != "" {
+				require.ErrorContains(t, err, tc.expectedErr)
```

Comment:
If you can do an errors.Is that compares a typed error to another that's ideal. Checking text can be fragile in the same way as checking console output. No change requested though, just something to keep in mind.

Reply:
I actually did decide to change it. The library (crane) doesn't expose these errors so I don't want tests to break if they change. In the future we can always update this if we have more test cases